flexmeasures/data/models/generic_assets.py

```diff
@@ -440,8 +440,9 @@ def search_beliefs(
                     append=True,
                 )
             df = df.reset_index()
-            df["source"] = df["source"].apply(lambda x: x.to_dict())
-            df["sensor"] = df["sensor"].apply(lambda x: x.to_dict())
+            if len(df) > 0:
```

Comment:
Doesn't this only affect the block starting in line 430? The source column seems to be unproblematic. Therefore, we could copy line 426 into said block as an alternative remedy?

Reply:
You're right! This only applies for the sensor.

Nonetheless, sensor in line 426 won't be defined as this happen when sensors = None.

I move the "source" out and also modified the condition to make it more explicit.